src/app/collection/settings/tab/rules/add-rule-form/blockly-form/blockly-editor/blockly-editor.component.ts

```diff
@@ -0,0 +1,792 @@
+/*
+ * Lumeer: Modern Data Definition and Processing Platform
+ *
+ * Copyright (C) since 2017 Answer Institute, s.r.o. and/or its affiliates.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {
+  AfterViewChecked,
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  EventEmitter,
+  Inject,
+  Input,
+  OnDestroy,
+  Output,
+  Renderer2,
+  ViewChild,
+} from '@angular/core';
+import {environment} from '../../../../../../../../environments/environment';
+import {Store} from '@ngrx/store';
+import {AppState} from '../../../../../../../core/store/app.state';
+import {ActivatedRoute} from '@angular/router';
+import {DialogService} from '../../../../../../../dialog/dialog.service';
+import {DOCUMENT} from '@angular/common';
+import {LinkType} from '../../../../../../../core/store/link-types/link.type';
+import {Collection} from '../../../../../../../core/store/collections/collection';
+import {Variable} from '../../../variable-type';
+import {shadeColor} from '../../../../../../../shared/utils/html-modifier';
+import {COLOR_DARK, COLOR_GRAY200, COLOR_GREEN, COLOR_PRIMARY, COLOR_RED} from '../../../../../../../core/constants';
+import {ContrastColorPipe} from '../../../../../../../shared/pipes/contrast-color.pipe';
+
+declare var Blockly: any;
+
+@Component({
+  selector: 'blockly-editor',
+  templateUrl: './blockly-editor.component.html',
+  styleUrls: ['./blockly-editor.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class BlocklyEditorComponent implements AfterViewInit, OnDestroy {
+  @Input('collections')
+  public collections: Collection[] = [];
+
+  @Input('linkTypes')
+  public linkTypes: LinkType[] = [];
+
+  @Input('variables')
```

Comment:
I think this is unnecessary..

Reply:
What do you mean by that? It is used. It will change in different use cases later (the component will likely become more generic).

Reply:
I mean string `'variables'` in input decorator. It's redundant.